src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -29,9 +29,10 @@ public class WebHostBuilder : IWebHostBuilder
         private readonly List<Action<ILoggerFactory>> _configureLoggingDelegates;
 
         private IConfiguration _config;
-        private ILoggerFactory _loggerFactory;
         private WebHostOptions _options;
         private bool _webHostBuilt;
+        private Func<HostingContext, ILoggerFactory> _createLoggerFactoryDelegate;
+        private Action<HostingContext, IConfigurationBuilder> _configureConfigurationBuilderDelegate;
```

Comment:
Lets do multiple. Its weird for one call to silently overwrite another. This way different components can contribute to config.